walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,74 @@
 
 package com.google.sps;
 
+import java.io.*;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+
 
 public final class FindMeetingQuery {
+  // array representing the number of meetings happening during that minute
+  private ArrayList<Integer> meetings;
+
+  public FindMeetingQuery() {
+    meetings = new ArrayList<Integer>();
```

Comment:
It's interesting what happens here. You are creating a class data structure called meetings.

Then every time you call query(), this structure gets modified. This will cause further calls to query() to behave unexpectedly, as each call to query mutates the state of this class.

This is generally an antipattern: a method called `query` should generally not mutate internal class state. I would suggest making `meetings` a local variable instead (created inside query() function and potentially passed to every helper which needs it).

Reply:
Thanks for noticing that, I will change `meetings` to be a local variable.